xgo/ios-simulator/update_ios.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+#
+# Contains a simple tool that updates some of the iOS toolchains with the SDKs
+# explicitly provided. The goal is to allow using your own up to date SDKs or
+# the simulator one not supported out of the box.
+#
+# Usage: update_ios.sh <path to>/<iSomething><Version>.sdk.tar.<type>
+set -e
+
+# Figure out the base name of the SDK
+sdk=`basename $1`
+sdk=${sdk%.*}
+sdk=${sdk%.*}
+
+# Define a small extraction utility to
+function extract {
+  case $1 in
+    *.tar.xz)
+      xz -dc $1 | tar xf -
+      ;;
+    *.tar.gz)
+      gunzip -dc $1 | tar xf -
+      ;;
+    *.tar.bz2)
+      bzip2 -dc $1 | tar xf -
+      ;;
+  esac
+}
+
+# Extract the SDK, patch it, clean it up and prep for bootstrapping
+extract $1
+
+if [[ "`basename $1`" =~ ^iPhoneSimulator ]]; then
+  echo "Patching iOS simulator SDK with missing libraries..."
+  ln -s $OSX_NDK_X86/SDK/$OSX_SDK/usr/lib/system/libsystem_kernel.dylib   $sdk/usr/lib/system/libsystem_kernel.dylib
+  ln -s $OSX_NDK_X86/SDK/$OSX_SDK/usr/lib/system/libsystem_platform.dylib $sdk/usr/lib/system/libsystem_platform.dylib
+  ln -s $OSX_NDK_X86/SDK/$OSX_SDK/usr/lib/system/libsystem_pthread.dylib  $sdk/usr/lib/system/libsystem_pthread.dylib
+  ln -s $OSX_NDK_X86/SDK/$OSX_SDK/usr/lib/system/libsystem_kernel.tbd     $sdk/usr/lib/system/libsystem_kernel.tbd
+  ln -s $OSX_NDK_X86/SDK/$OSX_SDK/usr/lib/system/libsystem_platform.tbd   $sdk/usr/lib/system/libsystem_platform.tbd
+  ln -s $OSX_NDK_X86/SDK/$OSX_SDK/usr/lib/system/libsystem_pthread.tbd    $sdk/usr/lib/system/libsystem_pthread.tbd
+fi
+
+tar -czf /tmp/$sdk.tar.gz $sdk
+rm -rf $sdk
+
+# Pull the iOS cross compiler tool and build the toolchain
+git clone https://github.com/tpoechtrager/cctools-port.git
+cd cctools-port
+git reset --hard adf616eee9d41f4961c3a83ba275249ffcb32d33
```

Comment:
FYI @PombeirP, I needed to pin down this dependency.

Reply:
Thanks @adambabik I was going to ask what had changed in order for this to start working again. You might want to pass `-n` to `git clone` so that it doesn't checkout `HEAD`, since you'll be resetting to a specific commit right afterwards.

Reply:
Thanks for the tip!